includes/Story_Renderer/Embed.php

```diff
@@ -63,32 +64,53 @@ public function __construct( Story $story ) {
 	 * @return string Rendered block type output.
 	 */
 	public function render( array $args = [] ) {
-		$defaults     = [
+		$defaults = [
 			'align'  => 'none',
 			'height' => 600,
 			'width'  => 360,
 		];
-		$args         = wp_parse_args( $args, $defaults );
-		$align        = sprintf( 'align%s', $args['align'] );
-		$url          = $this->story->get_url();
-		$title        = $this->story->get_title();
-		$poster       = ! empty( $this->story->get_poster_portrait() ) ? esc_url( $this->story->get_poster_portrait() ) : '';
-		$margin       = ( 'center' === $args['align'] ) ? 'auto' : '0';
-		$player_style = sprintf( 'width: %dpx;height: %dpx;margin: %s', absint( $args['width'] ), absint( $args['height'] ), esc_attr( $margin ) );
-		$poster_style = ! empty( $poster ) ? sprintf( '--story-player-poster: url(%s)', $poster ) : '';
 
-		ob_start();
+		$args   = wp_parse_args( $args, $defaults );
+		$align  = sprintf( 'align%s', $args['align'] );
+		$url    = $this->story->get_url();
+		$title  = $this->story->get_title();
+		$poster = ! empty( $this->story->get_poster_portrait() ) ? esc_url( $this->story->get_poster_portrait() ) : '';
+
+		$poster_style  = ! empty( $poster ) ? sprintf( '--story-player-poster: url(%s)', $poster ) : '';
+		$wrapper_style = sprintf(
+			'--aspect-ratio: %F; --width: %dpx; --height: %dpx',
+			$args['height'] / $args['width'],
+			absint( $args['width'] ),
+			absint( $args['height'] )
+		);
+
+		// This CSS is used for AMP and non-AMP.
+		wp_enqueue_style( Embed_Block::SCRIPT_HANDLE_FRONTEND );
```

Comment:
I don't like how this references another class and requires that class to be run for this to work. Don't you also need the method `init_assets` to run here. I think the contents of `init_assets` should be moved to trait so both classes can reference this repeated logic?

Reply:
`init_assets ` should not really be reusable / run twice though, so a trait isn't the best idea.

Scripts and styles only need to be registered **once**.

The `Embed` class is instantiated many times, but `Embed_Block` is only instantiated once in the `Plugin` class.

If anything, there could be a new `Assets` class or anything that registers all commonly used scripts/styles.

Then `Embed_Block` can enqueue just the scripts/styles it needs, and `Embed` enqueues the scripts/styles it needs.

Reply:
As mentioned in https://github.com/google/web-stories-wp/issues/3135 there is now a possiblity that there will be a third place where this script will be refenereced.

Reply:
Added a to-do note for this so we can split this out into its own class in a separate ticket & PR